src/browser-connection/gateway.js

```diff
@@ -64,6 +66,7 @@ export default class BrowserConnectionGateway {
     static onHeartbeat (req, res, connection) {
         if (BrowserConnectionGateway.ensureConnectionReady(res, connection)) {
```

Comment:
The best option will be to prevent caching in all handlers before `ensureConnectionReady` call. Because if it will respond with `500` it may be cached.